frontend/dockerui/build.go

```diff
@@ -50,6 +50,23 @@ func (bc *Client) Build(ctx context.Context, fn BuildFunc) (*ResultBuilder, erro
 			if tp != nil {
 				p = *tp
 			}
+
+			// in certain conditions we allow input platform to be extended from base image
+			if p.OS == "windows" && img.OS == p.OS {
+				if p.OSVersion == "" && img.OSVersion != "" {
+					p.OSVersion = img.OSVersion
+				}
+				if p.OSFeatures == nil && len(img.OSFeatures) > 0 {
+					p.OSFeatures = img.OSFeatures
+				}
+			}
+
+			if p.Architecture == "arm" && img.Architecture == p.Architecture {
```

Comment:
As discussed, we should probably remove the `arm` case :eyes:

Reply:
Removed. Yeah, it doesn't make sense to downgrade arm variant from the base image if the user didn't set it.